recipe_scrapers/allrecipes.py

```diff
@@ -1,5 +1,33 @@
+import inspect
+import logging
+import re
+
 from ._abstract import AbstractScraper
 
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+UNITIZERX = re.compile(r"^([0-9.]+)\s*([^0-9.]*)$")
```

Comment:
What is this regular expression matching?

Reply:
It's used by `nutrients_unitized`, and could be relocated there, doesn't need module scope

Reply:
Moved

Reply:
Thanks - is there another variable name we could use to make its purpose a bit clearer?

Reply:
`unitize_splitter_rx`?

In retro, the one reason for keeping it module or class-level is to avoid recompiling it each time (same for reinitializing `DICT_FIELDS` if moved to `to_dict`) -habit from coding for high-volume sites. Initialize once, use for service lifecycle.

Reply:
Good point re: repeat compilations.  Are you from a C/C++ background by any chance?  We could go back to using a constant (all-capitalized by convention) for this, and place it on the `AllRecipes` class to get the best of both worlds (compiled once and not at module-level).

I'd suggest `QUANTITY_REGEXP`, and also to use Python regex 'named groups' (see https://pymotw.com/2/re/ for an example) to retrieve two match groups called `amount` and `units`.

Reply:
Yeah, I've got some C/C++ in my coder DNA.

I'm going to leave this PR open for a bit, while I'm testing it, and will come back round to finish it once the knowledge that it works across all allrecipes recipes has been acquired..

Reply:
Update: I coded up a live testing harness, crawled allrecipes.com (AR) into mongodb, and like you'd expect discovered issues in the process, with most fixed.

Most are triggered by the new `.to_dict()` method, in that when it invokes `self.<attribute_method>()` or `self.schema.<attribute_method>()`, it occasionally encounters glitches both in data and framework that raise exceptions from the underlying project. Some of these are a recursion exception that occurs within the plugin decorator framework. I haven't taken time to understand that code well enough to debug it, so I put in place some try/except to defend against data issues (and log them) and finally got through one "clean" crawl. (Fwiw, across the approximately 74,000 recipes scraped, there are thousands of exceptions being caught and logged.)

The AR recipe set turns out to be nearly 50% spam (one recipe is repeated 36,000 times, which specifies [one brand-name food product as an ingredient](https://www.allrecipes.com/recipe/165512/unleavened-bread/), named complete with "®" -- smells like a hack), so I coded to filter that one out. The net of actual user-submitted recipes was about 38,000.

I mention all that because this PR would be much affected if I commit those changes onto it. I could a) apply fixes for the last things we discussed @jayaddison, and, after merging to main, submit another PR containing the commits needed to get it working across the crawl. Or, b) close this PR and open another with the newer dev branch (whose parent is this PR's dev branch) Merging the first PR #a will result in code that isn't functional across all valid AR URIs, but that state of the repo would only last for a short time.

The other thing is the crawler. It's based on `requests` and `multiprocessing` pools, with the storage backend somewhat abstracted. Option #b above makes more sense, for review, in the context of this crawler code. The crawler is kind of rudimentary and not fully guardrailed, although you can kill it and it'll pick up where it left off, more or less. It parallelizes nicely, though I figure cloudflare rate-limited me more than once (AR's CDN). It's dockerized, with the goal of running it more stealthily in a datacenter VM with some IPv4 diversity.

To my point: the crawler is currently a separate project from `recipe-scrapers`: I'm calling it `recipe-crawler` for lack of imagination. I can (and will) put it up at the same time as the completion of this PR (either approach #a or #b), but wonder if adding it to this project is worth consideration, but worth pointing out that there is a dependency, be it one-way at present.

**Thoughts on the above, by you project maintainers, would be appreciated.**

Sidebar: I didn't find any Python crawler frameworks that made me happy (if you know of any, do tell), but also I had some odd requirements, such as caching all the http/https traffic to disk (to make recrawls less of a nuisance for site admins, and faster testing runtimes for myself), plus abstracting out the particular scraper framework and backend store. As this project lacks a corporate context, or jira/agile context, I went whiteboard and kanban for a day.